File: lib/diet_tracker_backend/models/recipe.js
const mongoose = require('mongoose');

const recipeSchema = new mongoose.Schema({
    name: { type: String, required: true },
    description: { type: String, required: true },
    calories: { type: Number, required: true },
    proteins: { type: Number, required: true },
    fats: { type: Number, required: true },
    carbs: { type: Number, required: true },
    grams: { type: Number, required: true },
    photo: { type: String }, // URL to the photo
    videoLink: { type: String } // URL to the video
}, { timestamps: true });

module.exports = mongoose.model('Recipe', recipeSchema);

File: lib/features/recpies/data/recipe.dart
class Recipe {
  final String name;
  final String image;
  final int calories;
  final int protein;
  final int fat;
  final int carbs;
  final int time;
  final String description;
  final String? videoUrl; // Nullable video URL

  Recipe({
    required this.name,
    required this.image,
    required this.calories,
    required this.protein,
    required this.fat,
    required this.carbs,
    required this.time,
    required this.description,
    this.videoUrl, // Optional parameter
  });

  static fromJson(e) {}
}
